src/core/cuda_utils.cc

```diff
@@ -87,7 +96,14 @@ CopyBuffer(
   // the src buffer is valid.
   if ((src_memory_type != TRITONSERVER_MEMORY_GPU) &&
       (dst_memory_type != TRITONSERVER_MEMORY_GPU)) {
-    memcpy(dst, src, byte_size);
+    if (zero_copy_support) {
+      auto params = new CopyParams(dst, src, byte_size);
```

Comment:
Considering that the synchronization will be done in `Finalize()`, in other words, the responder object will be valid during the execution of the host function, you don't need to `new / delete` the params object: it can just be a member of responder and you pass the pointer to the member as usrp.

Reply:
Sure, I will try a build without the new/delete for params

Reply:
I verified that the new/delete is indeed needed.

Reply:
Why? Responder is deleted before host function returns?

Reply:
The lifecycle of the CopyParams is such that the object is destroyed before the memcpy call inside the cuda host function

Reply:
My mistake, it doesn't reside in responder